types/react-big-calendar/index.d.ts

```diff
@@ -148,29 +148,29 @@ export interface ResourceHeaderProps {
 export interface Components<TEvent extends object = Event> {
     event?: React.ComponentType<EventProps<TEvent>>;
     eventWrapper?: React.ComponentType<EventWrapperProps<TEvent>>;
-    eventContainerWrapper?: React.SFC | React.Component | React.ComponentClass | JSX.Element;
-    dateCellWrapper?: React.SFC | React.Component | React.ComponentClass | JSX.Element;
-    timeSlotWrapper?: React.SFC | React.Component | React.ComponentClass | JSX.Element;
-    timeGutterHeader?: React.SFC | React.Component | React.ComponentClass | JSX.Element;
-    timeGutterWrapper?: React.SFC | React.Component | React.ComponentClass | JSX.Element;
+    eventContainerWrapper?: React.ComponentType;
+    dateCellWrapper?: React.ComponentType;
+    timeSlotWrapper?: React.ComponentType;
+    timeGutterHeader?: React.ComponentType;
+    timeGutterWrapper?: React.ComponentType;
     toolbar?: React.ComponentType<ToolbarProps>;
     agenda?: {
-        date?: React.SFC | React.Component | React.ComponentClass | JSX.Element;
-        time?: React.SFC | React.Component | React.ComponentClass | JSX.Element;
-        event?: React.SFC | React.Component | React.ComponentClass | JSX.Element;
+        date?: React.ComponentType;
+        time?: React.ComponentType;
+        event?: React.ComponentType<EventProps<TEvent>>;
     };
     day?: {
-        header?: React.SFC | React.Component | React.ComponentClass | JSX.Element;
-        event?: React.SFC | React.Component | React.ComponentClass | JSX.Element;
+        header?: React.ComponentType;
+        event?: React.ComponentType<EventProps<TEvent>>;
     };
     week?: {
-        header?: React.SFC | React.Component | React.ComponentClass | JSX.Element;
-        event?: React.SFC | React.Component | React.ComponentClass | JSX.Element;
+        header?: React.ComponentType;
+        event?: React.ComponentType<EventProps<TEvent>>;
     };
     month?: {
-        header?: React.SFC | React.Component | React.ComponentClass | JSX.Element;
-        dateHeader?: React.SFC | React.Component | React.ComponentClass | JSX.Element;
-        event?: React.SFC | React.Component | React.ComponentClass | JSX.Element;
+        header?: React.ComponentType;
```

Comment:
This no longer allows elements. Was this wrong before?

Reply:
Yes, according to my tests, assigning elements doesnt work:

https://codesandbox.io/s/react-big-calendar-example-sygd2?file=/index.js

(But I did not check the source...)

Reply:
Implementation access elements from components like this:

```typescript
const { event: Event, date: AgendaDate } = components
// ...
```

```jsx
<td className="rbc-agenda-event-cell">
  {Event ? <Event event={event} title={title} /> : title}
</td>
```

And the demos in `examples/demos`  that are specifying a `components` prop are only specifying functional components.